tests/test_scalars.py

```diff
@@ -156,116 +181,131 @@ def test_timedelta_add_sub() -> None:
-    check(assert_type(td + as1, pd.Period), pd.Period)
-    check(assert_type(td + as2, pd.Timestamp), pd.Timestamp)
-    check(assert_type(td + as3, pd.Timestamp), pd.Timestamp)
-    check(assert_type(td + as4, dt.date), dt.date)
-    check(assert_type(td + as5, pd.Timestamp), pd.Timestamp)
-    check(assert_type(td + as6, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td + as7, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td + as8, pd.TimedeltaIndex), pd.TimedeltaIndex)
-    check(assert_type(td + as9, pd.Series), pd.Series)
-    check(assert_type(td + as10, pd.PeriodIndex), pd.PeriodIndex)
-    check(assert_type(td + as11, pd.DatetimeIndex), pd.DatetimeIndex)
-    check(assert_type(td + as12, npt.NDArray[np.timedelta64]), np.ndarray)
+    check(assert_type(td + as_period, pd.Period), pd.Period)
+    check(assert_type(td + as_timestamp, pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + as_datetime, pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + as_date, dt.date), dt.date)
+    check(assert_type(td + as_datetime64, pd.Timestamp), pd.Timestamp)
+    check(assert_type(td + as_dt_timedelta, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td + as_timedelta64, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td + as_timedelta_index, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td + as_timedelta_series, TimedeltaSeries), pd.Series)
+    check(assert_type(td + as_period_index, pd.PeriodIndex), pd.PeriodIndex)
+    check(assert_type(td + as_datetime_index, pd.DatetimeIndex), pd.DatetimeIndex)
+    check(
+        assert_type(td + as_ndarray_td64, npt.NDArray[np.timedelta64]),
+        np.ndarray,
+        np.timedelta64,
+    )
     # pyright has trouble with timedelta64 and datetime64
     check(
         assert_type(
-            td + as13,  # pyright: ignore[reportGeneralTypeIssues]
+            td + as_ndarray_dt64,  # pyright: ignore[reportGeneralTypeIssues]
             npt.NDArray[np.datetime64],
         ),
         np.ndarray,
+        np.datetime64,
     )
-    check(assert_type(td + as14, NaTType), NaTType)
-
-    check(assert_type(as1 + td, pd.Period), pd.Period)
-    check(assert_type(as2 + td, pd.Timestamp), pd.Timestamp)
-    check(assert_type(as3 + td, dt.datetime), dt.datetime)
-    check(assert_type(as4 + td, dt.date), dt.date)
-    check(assert_type(as5 + td, pd.Timestamp), pd.Timestamp)
-    # pyright is wrong here because as6 + td calls td.__radd__(as6),
-    # not timedelta.__add__
+    check(assert_type(td + as_nat, NaTType), NaTType)
+
+    check(assert_type(as_period + td, pd.Period), pd.Period)
+    check(assert_type(as_timestamp + td, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as_datetime + td, dt.datetime), dt.datetime)
+    check(assert_type(as_date + td, dt.date), dt.date)
+    check(assert_type(as_datetime64 + td, pd.Timestamp), pd.Timestamp)
+    # pyright can't know that as6 + td calls td.__radd__(as6),  not timedelta.__sub__
```

Comment:
change comment to say `as_dt_timedelta` instead of `as6` and you mean `timedelta.__add__` here not `timedelta.__sub__`

Reply:
Fixed.